File: 243_Shortest_Word_Distance.py

class Solution(object):
    def shortestDistance(self, words, word1, word2):
        p1 = p2 = -1
        res = len(words)
        for i, word in enumerate(words):
            if word == word1: 
                p1 = i
            elif word == word2: 
                p2 = i
            if p1 >= 0 and p2 >= 0:
                res = min(res, abs(p1 - p2))
        return res

File: 243_Shortest_Word_Distance.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution{
public:
	int shortestDistance(vector<string>& words, string word1, string word2){
		int res = words.size(), p1 = -1, p2 = -1;
		for (int i = 0; i < words.size(); ++i){
			if (words[i] == word1) p1 = i;
			else if (words[i] == word2) p2 = i;
			if (p1 >= 0 && p2 >= 0) res = min(res, abs(p1 - p2));
		}
		return res;
	}
};

int main(){
	Solution* solution = new Solution();
	vector<string> words = {"practice", "makes", "perfect", "coding", "makes"};
	cout << solution->shortestDistance(words, "coding", "practice") << "\n";
	return 0;
}